### DISCUSSION TITLE
Renaming suggestions

### DISCUSSION THREAD

**Discussion by EmiOnGit** [EmiOnGit/warbler_grass#17] [Q&A]
I am totally aware that some names from structures became misleading over many early iterations, however I am not very good at naming.

This thread is supposed to be a place to share your thoughts.
I would like to have your thoughts on,
which names you find misleading and would like to have renamed.

If you don't know a better name yourself this is totally fine.
Sharing what is wrong is more than not sharing because you don't have a solution

**Comment by EmiOnGit** [comment]
`RegionConfiguration` the name is misleading
A game can only have one`RegionConfiguration` which in itself is contradictory. The user would think he can define multiple regions in his game.
As for suggestions.
Maybe just `WarblersConfiguration` or `GrassConfiguration`

**Comment by janhohenheim** [comment]
From the other chat:
> `warblersneeds` to me reads like a single fictional word, so I had some trouble remembering the name until I realized it was supposed to be read as `warblers_needs` 🙂

**Reply by janhohenheim** [reply]
[`warbler` is also available on crates.io](https://crates.io/search?q=warbler)

**Reply by EmiOnGit** [reply]
Warbler would be also great.
My thoughts:
* The usecase of the crate wouldn't be obvious from the name. There are other crates which doesn't care about this to much however ( hanabi, rapier)
* searching for warbler in the internet would just result in pictures of birds. Depending on how much the crate is used, even warbler rust might not work always

However I still am intrigued to use the name

**Reply by EmiOnGit** [reply]
It's also possible to ask for a name with bevy as prefix such as bevy_grass

**Reply by janhohenheim** [reply]
Big fan of `bevy_grass`! 😄

**Reply by EmiOnGit** [reply]
I like it too, would be a big responsibility since there can only be one `bevy_grass` though😄 
`bevy_warbler` is open too but I'd prefer `warbler` over `bevy_warbler`

**Reply by EmiOnGit** [reply]
Update on the names: 
`bevy_grass` should be avoided since it sounds to official and might mislead others
`bevy_warbler` would be fine however. 
Also `warbler_grass` was suggested, as it is more distinctive

**Reply by janhohenheim** [reply]
Uuuh I like `warbler_grass`!

**Reply by EmiOnGit** [reply]
Same. What about `warblers_grass` though? Gramatically it'd make more sense

**Reply by EmiOnGit** [reply]
Any preferences at `warblers_grass` vs `warbler_grass`?

**Reply by janhohenheim** [reply]
Slight preference for `warbler_grass`, as in the "the grass for warblers", because the missing letter makes the crate name simpler in my mind. But go with whatever your gut tells you :)
